modules/javafx.media/src/main/native/gstreamer/plugins/mfwrapper/mfwrapper.cpp

```diff
@@ -789,7 +757,7 @@ static gint mfwrapper_process_output(GstMFWrapper *decoder)
         return PO_FLUSHING;
 
     HRESULT hr = decoder->pDecoder->GetOutputStatus(&dwFlags);
-    if (dwFlags != MFT_OUTPUT_STATUS_SAMPLE_READY)
+    if (SUCCEEDED(hr) && dwFlags != MFT_OUTPUT_STATUS_SAMPLE_READY)
```

Comment:
If the previous call failed, should you really continue with the following call to `ProcessOutput`?